### PR TITLE
feat: upgraded connectivity_plus package

### PR DESCRIPTION
**Pull request by Avinash-dhiraj**
- As app store introduced mandatory privacy manifest files for some of packages being used in the application. connectivity_plus 6.0.2 is one of them and current version of connectivity_plus 6.0.2 has added support for privacy manifest but graphql_flutter still using connectivity_plus 5.0.0 which leads into compatibility issue. App store is very specific about the deadline of these as well which is 1st of may.;

upgraded connectivity_plus from 5.0.0 to 6.0.2

Please merge this on priority as this breaking changes is needed for all the flutter application which is using connectivity_plus 6.0.2 along with graphql_flutter

Fixes https://github.com/zino-hofmann/graphql-flutter/issues/1423#issuecomment-2052495874

### CODE REVIEW

**Comment by vincenzopalazzo** [comment]
looks like that needs a rebase, please ping me when the conflict are fixed

**Comment by Avinash-dhiraj** [comment]
> looks like that needs a rebase, please ping me when the conflict are fixed

there is one conflict in pubspec file 
```
<<<<<<< connectivity-plus-upgrade
  connectivity_plus: ^6.0.2
=======
  connectivity_plus: ^5.0.0
>>>>>>> main
```
which can be resolved when someone merge it.

File: packages/graphql_flutter/lib/src/widgets/hooks/subscription.dart [hunk 1]
```diff
@@ -69,8 +69,9 @@ class _SubscriptionHookState<TParsed> extends HookState<
     Stream<QueryResult<TParsed>>, _SubscriptionHook<TParsed>> {
   late Stream<QueryResult<TParsed>> stream;
 
-  ConnectivityResult? _currentConnectivityResult;
```
Comment (vincenzopalazzo) [inline packages/graphql_flutter/lib/src/widgets/hooks/subscription.dart:72]: this commit header should start with `fix(graphel_flutter):`

File: packages/graphql_flutter/pubspec.yaml [hunk 2]
```diff
@@ -15,6 +15,7 @@ dependencies:
   meta: ^1.7.0
   path_provider: ^2.0.1
   path: ^1.8.0
+  #connectivity_plus: ^5.0.0
```
Comment (vincenzopalazzo) [inline packages/graphql_flutter/pubspec.yaml]: not sure what you want to do with this commit, but I think this should be folded in the previous one

File: packages/graphql_flutter/pubspec.yaml [hunk 3]
```diff
@@ -15,7 +15,6 @@ dependencies:
   meta: ^1.7.0
   path_provider: ^2.0.1
   path: ^1.8.0
-  #connectivity_plus: ^5.0.0
```
Comment (vincenzopalazzo) [inline packages/graphql_flutter/pubspec.yaml]: same, it need to be folded with the previous one

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
We do not use merge commit inside the PR, so you should rebase it with `git rebase`.

This commit can be removed dee9b9414b8ae275b541174aaeaa61ff39744d70

In addition I left a couple of comments because I am a little bit confused while looking at your commit history

**Comment by johnmartel** [comment]
@vincenzopalazzo any chance of getting this merged and released quickly if I fork this PR and submit an updated version? This is creating an unnecessary roadblock for projects that use both `graphql_flutter` and `connectivity_plus` that want to upgrade their dependencies.

**Comment by selawik** [comment]
Any updates?

**Comment by SachinCodeMatrix** [comment]
@Avinash-dhiraj @vincenzopalazzo  please check this

**Comment by vincenzopalazzo** [comment]
This is unresolved https://github.com/zino-hofmann/graphql-flutter/pull/1424#pullrequestreview-1999240159

I have nothing to check

**Comment by vincenzopalazzo** [comment]
Closing because looks like that there is already a fix https://github.com/zino-hofmann/graphql-flutter/issues/1423#issuecomment-2134515753

Please open a new PR with a good git history if the problem persit

**Pull request closed without merging**